src/handlers/custom-options.js

```diff
@@ -9,17 +11,23 @@ function customOptionsHandler(options, next) {
         headerGeneratorOptions,
         useHeaderGenerator,
         context,
-        ...gotOptions } = options;
+    } = options;
 
     // Got expects custom properties inside the context option.
-    gotOptions.context = {
+    const newContext = {
         ...context,
         proxyUrl,
         headerGeneratorOptions,
         useHeaderGenerator,
     };
 
-    return next(gotOptions);
+    delete options.proxyUrl;
+    delete options.headerGeneratorOptions;
+    delete options.useHeaderGenerator;
```

Comment:
Aren't you deleting this from the object the user provided?

Reply:
It should be already a new object provided by got. It is already merged with the defaults.